pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -310,6 +361,93 @@ func (e *TokensController) syncSecret() {
 	}
 }
 
+func (e *TokensController) shouldCleanToken(secret *v1.Secret, sa *v1.ServiceAccount) bool {
+	if !e.legacyTokenCleanUp || e.legacyTokenCleanUpPeriod == 0 {
```

Comment:
you could base this check solely on `e.legacyTokenCleanUpPeriod == 0` if you don't set the `TokensController` struct attribute's value when the feature gate is off

Reply:
Yeah. I previously planned to do that. But as Jordan suggested, it is better to create a new controller to do the clean up, thus it can be separated with the existing code. I am currently working on revising the PR. Thank you for your comments, Stanislav!